integration-tests/src/endpoints/datasets.py

```diff
@@ -0,0 +1,140 @@
+            )
+
+            # Map each future to its feed ID
+            future_to_feed_id = {
+                executor.submit(self.validate_feed, feed_id): feed_id
+                for feed_id in feed_ids
+            }
+
+            for future in as_completed(future_to_feed_id):
+                feed_id = future_to_feed_id[future]
+                error_entry = future.result()
+                if error_entry is not None:
+                    errors.append(error_entry)
+                    self.console.log(
+                        f"Feed '{error_entry['stable_id']}' has a related [red]error[/red]: "
+                        f"{error_entry['Error Details']}"
+                    )
+                else:
+                    self.console.log(
+                        f"Feed 'mdb-{feed_id}' has a valid latest dataset :white_check_mark:"
+                    )
+
+                # Update the progress bar
+                self.progress.update(task, advance=1)
+
+        if errors:
+            # If there were errors, log them as before
+            self._log_errors(errors)
+
+    def validate_feed(self, feed_id):
+        stable_id = f"mdb-{feed_id}"
+        error_detail = None
+        status_code = None
+        response = None
+        try:
+            # Validate feed response
+            response = self.get_response(f"v1/feeds/{stable_id}")
+            self._validate_feed_response(response)
+
+            # Validate datasets
+            response = self.get_response(
+                f"v1/gtfs_feeds/{stable_id}/datasets", params={"latest": True}
+            )
+            datasets = response.json()
+            self._validate_dataset(datasets, response.status_code)
+        except Exception as e:
+            error_detail = str(e)
+            if "response" in locals():
+                status_code = response.status_code
+
+            # Instead of re-raising the exception, return the error entry directly
+            return self._create_error_entry(stable_id, error_detail, status_code)
+
+        # If the try block completes without exceptions, return None to indicate no error
+        return None
+
+    @staticmethod
+    def _validate_feed_response(response):
+        if response.status_code != 200:
+            error_code = (
+                FeedError.NOT_FOUND.name
+                if response.status_code == 404
+                else FeedError.API_ERROR.name
+            )
+            error_detail = (
+                FeedError.NOT_FOUND.value
+                if response.status_code == 404
+                else FeedError.API_ERROR.value
+            )
+            raise Exception(f"{error_code}: {error_detail}")
+
+    @staticmethod
+    def _validate_dataset(datasets, status_code):
+        if status_code != 200 or not datasets:
+            error_code = FeedError.NO_DATASET.name
+            error_detail = FeedError.NO_DATASET.value
+            raise Exception(f"{error_code}: {error_detail}")
```

Comment:
I'm not sure if no having a dataset is an error. It's a valid use case.

Reply:
The exception is caught and doesn't make the script fail! It's just to create the report and express that this feed does not have a dataset

Reply:
I see. Is this also the case with the feed validation?

Reply:
Updated: the code throws an exception after all test have ran if any failed! (datasets validation excluded)